### PR TITLE
[ADD] 7.0 Salary rule reference

### PR DESCRIPTION
**Pull request by ghost**
Split module salary rule variable in 2 modules.

Module hr_salary_rule_reference allows to reference the rule itself by its id or a browse record.
Module hr_salary_rule_variable uses the browse record to query the variables with the method variable.

### CODE REVIEW

**Comment by ghost** [comment]
@bwrsandman @max3903 Can you review this PR please.

File: hr_salary_rule_variable/hr_salary_rule.py [hunk 1]
```diff
@@ -32,13 +34,50 @@ class hr_salary_rule(orm.Model):
...
+        # requested period
+        variable_list = [
+            variable for variable in rule.variable_ids
+            if variable.date_from <= date <= variable.date_to
+        ]
+        if not variable_list:
+            raise orm.except_orm(
+                _("Warning"),
+                _("The salary rule variable related to %s does not "
+                    "exist for the date %s""") %
```
Comment (bwrsandman) [inline hr_salary_rule_variable/hr_salary_rule.py]: What's with the odd spacing here?

File: hr_salary_rule_variable/hr_salary_rule.py [hunk 2]
```diff
@@ -32,13 +34,50 @@ class hr_salary_rule(orm.Model):
...
+                _("Warning"),
+                _("%s salary rule variables related to %s exist for "
+                    "the date %s""") %
+                (len(variable_list), rule.code, date))
+
+        variable = variable_list[0]
+
+        # Return the result whether the variable is fix or based on python code
+        if variable.type == 'python':
+            return eval(variable.python_code, localdict or {})
```
Comment (bwrsandman) [inline hr_salary_rule_variable/hr_salary_rule.py:81]: Is it safe to execute arbitrary python code?

Comment (ghost) [inline hr_salary_rule_variable/hr_salary_rule.py:81]: @bwrsandman I think so, as long as the unit tests cover every rule that is computed this way.

File: hr_salary_rule_variable/hr_salary_rule.py (hunk 1 above)
Comment (ghost) [inline hr_salary_rule_variable/hr_salary_rule.py]: will fix

**Comment by bwrsandman** [comment]
:+1:

**Comment by max3903** [comment]
:+1:

**Pull request merged**